### ISSUE TITLE
Instruction for running 

### ISSUE THREAD

**Issue by lalithr95** [abdulla-alali/CRE-NS3#1]
Documentation link in the README doesn't work. How can get the steps for running this locally ?

Thanks !

**Comment by ghost** [comment]
Try downloading the CR patch through git. To do it, download ns-3.17 (preferred) from the repository, do not build it yet, cd into the ns-allinone-3.17 directory and remove the ns-3.17 directory. Now being on the same directory, download the git package of CRE-NS3 provided in the link https://github.com/abdulla-alali/CRE-NS3/ , later build the ns-3 using configure and build commands. Try building without --enable-tests --enable-examples for clean build without error. Hope this helps.

**Comment by jcx70100** [comment]
Why i changed the datas in the "map_PUs_multiple.txt" file, the location of nodes doesn't change? Is there any rules of the data?

Thanks!

**Comment by ghost** [comment]
Hello,

Thanks for the resources, I found them to be very informative for my
research. I have understood the Map File data, but are the Arrival times
and Departure times are manual entry, or auto generated ?

Thank you.
On May 1, 2016 6:03 PM, "jcx70100" notifications@github.com wrote:

Why i changed the datas in the "map_PUs_multiple.txt" file, the location of
nodes doesn't change? Is there any rules of the data?

Thanks!

**Comment by jcx70100** [comment]
Thanks, i under the first three lines of data,but when i changed it, it doesn't work.When i clear the data of the Map File, the result seems the same. I still can't find out the rule of the data under the "21",but i really means the departure and arrival times.

Thank you

**Comment by ghost** [comment]
Thanks for the information. I am understanding the concept of this CRN
model very well.

Thank you.
On May 1, 2016 6:22 PM, "jcx70100" notifications@github.com wrote:

Thanks, i under the first three lines of data,but when i changed it, it
doesn't work.When i clear the data of the Map File, the result seems the
same. I still can't find out the rule of the data under the "21",but i
really means the departure and arrival times.

Thank you

**Comment by ghost** [comment]
Hello,

I have been performing evaluations for the example code given in your blog.
I generated a Net Animator XML file and executed it. Why does it only shows
2 nodes with a path and region signals ? Are those Secondary User Nodes or
Primary Users ? And I don't understand under what spectrum range both SU
and PU works ? I read through the code and it defines 802..11g somewhere
and speedlink of 54Mbps. Are those for Seondary Users ? What range does PU
utilizes ?

Any information could be helpful.

Thank you

**Comment by jcx70100** [comment]
Hi

I`m sorry for replying you after a long time.
I`m now studying the CRE-NS3 code, I have to say, I`m not very skilled at ns3. The code given in the website only set 2 nodes, when I compiled the code, we can only find tow nodes in NetAnim.
Moreover,the coordinates of the nodes do not match those setted in the “txt”file.
How about your study of this code?
Do you know how to control the coordinates of the nodes?

Maybe we can communicate about this code.

Thanks for your mail.

发件人: bharaniamarnath [mailto:notifications@github.com]
发送时间: 2016年5月8日 15:49
收件人: abdulla-alali/CRE-NS3 CRE-NS3@noreply.github.com
抄送: jcx70100 jcx70100@live.cn; Comment comment@noreply.github.com
主题: Re: [abdulla-alali/CRE-NS3] Instruction for running (#1)

Hello,

I have been performing evaluations for the example code given in your blog.
I generated a Net Animator XML file and executed it. Why does it only shows
2 nodes with a path and region signals ? Are those Secondary User Nodes or
Primary Users ? And I don't understand under what spectrum range both SU
and PU works ? I read through the code and it defines 802..11g somewhere
and speedlink of 54Mbps. Are those for Seondary Users ? What range does PU
utilizes ?

Any information could be helpful.

Thank you

**Comment by NaziaTrina** [comment]
If i do it without --enable-tests --enable-examples then how its gonna work ? @bharaniamarnath 
As i can't test it , and i can't run any of them . @bharaniamarnath

**Comment by ghost** [comment]
@NaziaTrina I remember that I tried to rebuild NS3 with --enable-tests --enable-examples after integrating the CRN-NS3 module, but it failed. Was the build successful for you after integrating CRN-NS3 and including the 'examples' and 'tests' ?

**Comment by NaziaTrina** [comment]
Yes, build was successful but it only can run the scratch cc file . I can't run the cognitive module.

**Comment by ghost** [comment]
@NaziaTrina The cc files in the 'cognitive' module can't be run directly or as a copy from scratch using waf. The module classes can be used as extension in a new cc file created in scratch or use the example.cc ( http://krc.coe.neu.edu/sites/krc.coe.neu.edu/files/papers/simulator/examples/example.cc ).

**Comment by NaziaTrina** [comment]
Thanks ! @bharaniamarnath

**Comment by tribbianii** [comment]
Hello,

Thank you for your support for CR extension in NS3. I've been playing with this code for couple of days and here are some questions I have about it. First, I tried to change the number of nodes when I run the example.cc, but it turns out that when there are more than 3 nodes the bytes received will always be 0. Why? I though given the certain number of PUs if we have more nodes each one would be more likely to receive packages from others; Secondly, how do we determine the arrival time and departure time in map file? It looks random and can we extend the time for PU activity in that file? Thirdly, in NS3 do we assign PUs in nodes? In example.cc we use InstallCR to all nodes. But for each PUs we do have specific coordinates in map file which are different with those of nodes we assign in example.cc, how does NS3 handle this? 

Thank you for presenting such extension and I do expect your reply. Appreciate it！

**Comment by abhimishra6396** [comment]
Hello,

I am using gcc 4.9. Compilation is okay but this is giving me following linking error on Mac. I don't know why cognitive module is showing dependencies on other modules like wifi and internet; which is giving errors. I would appreciate any suggetions:)
![Screenshot 2019-03-27 at 11 00 42 PM](https://user-images.githubusercontent.com/16163948/55115358-97241280-50e4-11e9-9845-37fc313ba6a0.png)
